### PR TITLE
Use pinned version of Hugo

### PR DESCRIPTION
**Pull request by rlenferink**
As explained in https://github.com/apache/comdev-site/pull/99#issuecomment-1490626581 not pinning tooling to a specific version is not too good of an idea in general. Having to rely on different tooling versions is something I experience on a daily basis and I would advise for Comdev to not start relying on tooling which different projects have different interest in. This to prevent accidentally breaking our site (either with a hard failure or a soft failure where the build succeeds but wrong output is generated).

This adds back the Hugo download (copied from the Pagefind part, so including checksum verification).

/cc @sebbASF @bdelacretaz 

EDIT: Jenkins build output:

```
+ hugo version
hugo v0.111.3-5d4eb5154e1fed125ca8e9b5a0315c4180dab192+extended linux/amd64 BuildDate=2023-03-12T11:40:50Z VendorInfo=gohugoio
[Pipeline] sh
+ pagefind --version
pagefind 0.12.0
```

See https://ci-builds.apache.org/job/Community%20Development/job/site/job/use-fixed-tooling-versions/1/console

### CODE REVIEW

**Review by bdelacretaz — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -56,10 +68,13 @@ pipeline {
         stage('Build') {
             steps {
                 script {
-                    sh 'for i in $(which -a hugo) ; do echo $i; $i version; done'
-                    sh 'hugo version'
-                    sh "hugo --destination ${env.OUT_DIR}"
-                    sh "${env.PAGEFIND_DIR}/bin/pagefind --source ${env.OUT_DIR}"
+                    withEnv(["PATH+HUGO=${env.HUGO_DIR}/bin", "PATH+PAGEFIND=${env.PAGEFIND_DIR}/bin"]) {
```
[the remark this answers is not in the document]
Comment (rlenferink) [reply Jenkinsfile:71]: Without changing the PATH and calling `hugo`, the system default installation is used.

Without changing the PATH and calling `${env.HUGO_DIR}/bin/hugo` the Hugo downloaded in the Jenkinsfile is used.

With changing the PATH, and calling `hugo` the Hugo downloaded in the Jenkinsfile is used. The `PATH+TOOLNAME` will ensure the specified path is prepended to the PATH (so a first search location). See https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#withenv-set-environment-variables 

The following would be the same:

```groovy
script {
    withEnv(["PATH+HUGO=${env.HUGO_DIR}/bin", "PATH+PAGEFIND=${env.PAGEFIND_DIR}/bin"]) {
        sh "hugo version"
        sh "pagefind --version"

        sh "hugo --destination ${env.OUT_DIR}"
        sh "pagefind --source ${env.OUT_DIR}"
    }
}
```

```groovy
script {
    sh "${env.HUGO_DIR}/bin/hugo version"
    sh "${env.PAGEFIND_DIR}/bin/pagefind --version"

    sh "${env.HUGO_DIR}/bin/hugo --destination ${env.OUT_DIR}"
    sh "${env.PAGEFIND_DIR}/bin/pagefind --source ${env.OUT_DIR}"
}
```

From a readability (and don't repeat yourself) perspective my preference is with the first code block (using `withEnv`).

**Comment by rlenferink** [comment]
> Surely the version of Hugo provided by Jenkins is effectively pinned?

It indeed is pinned. But if INFRA (or another project) requests the version to be updated then we might be lucky and it keeps working, but the newer version could also break things.

If it were installed using the Jenkins toolbox and Jenkins would allow for a Hugo type, then the following would IMO also be good:

```
tools {
  hugo 'Hugo 0.111.3'
}
```

But it seems that only works for maven, jdk and gradle: https://www.jenkins.io/doc/book/pipeline/syntax/#supported-tools.

**Comment by rlenferink** [comment]
> > But it seems that only works for maven, jdk and gradle: https://www.jenkins.io/doc/book/pipeline/syntax/#supported-tools.
> 
> However https://www.jenkins.io/doc/book/managing/tools/ says that the built-in tools are: Ant, Git, JDK and Maven. This suggests that it may be possible to configure other tools. Unfortunately there is no documentation.

The page you found seems to be more correct (checking my local Jenkins instance). But indeed, no documentation doesn't really help here.

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (rlenferink) [reply Jenkinsfile:71]: > If for some reason the hugo executable is not present at the first item on the path, then the withEnv version will fall back to checking the rest of the path. That would not be immediately obvious unless the build failed.

Good catch! With that in mind, I'll update the PR.

**Comment by rlenferink** [comment]
@sebbASF does this look okay to you to be merged?

**Comment by rlenferink** [comment]
They are used once in the `${HUGO} version` and once in the actual `${HUGO} --destination` call. Same for pagefind.

**Comment by rlenferink** [comment]
> however the version calls were added for debug purposes and are not needed

Removed those statements and simplified the call to Hugo (per your suggestions)

**Pull request merged**